code/modules/power/apc.dm

```diff
@@ -681,7 +681,7 @@
 		"chargingStatus" = charging,
 		"totalLoad" = lastused_total,
 		"coverLocked" = coverlocked,
-		"siliconUser" = user.has_unlimited_silicon_privilege,
+		"siliconUser" = user.has_unlimited_silicon_privilege || locate(/obj/machinery/computer/apc_control) in range(1, user),
```

Comment:
Ah, that could work. I don't like global procs, but hey! This thing might have some more uses added eventually, so I guess so.